soundbay/utils/conf_validator.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional
+from pydantic import BaseModel, validator
+
+
+class Dataset(BaseModel):
+    # label_names: ListConfig
+    batch_size: int
+    num_workers: int
+    sample_rate: int
+    data_sample_rate: int
+    max_freq: int
+    min_freq_filtering: int 
+    n_fft: int
+    hop_length: int
+    train_dataset: dict
+    val_dataset: dict
+    class Config:
+        title = "Dataset"
+        allow_mutation = False
+        validate_assignment = True
+        anystr_lower = False
+        validate_all = True
+        use_enum_values = True
+
+    @validator("num_workers")
+    def validate_num_workers(cls, num_workers:int):
+        if num_workers > 4:
+            raise ValueError(f"Number of works is larger than 4 {num_workers}")
```

Comment:
any reason to choose 4 as the maximum?

Reply:
No, it was just a check, nothing specific.